tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles.h

```diff
@@ -178,16 +211,30 @@ class TrtShapeOptimizationProfile {
   // Restores profiles from the engine (used after deserialization).
   Status RestoreProfiles(const nvinfer1::ICudaEngine* engine);
 
+  // Whether the network has any shape tensors.
+  bool HasShapeTensor() const { return has_shape_tensor_; }
+
+  void SetShapeTensorMask(const nvinfer1::INetworkDefinition* network);
+
  private:
   // Set of input shape vetors that we collect during profile_generation_mode.
-  std::unordered_set<std::vector<TensorShape>, VectorTensorShapeHasher>
-      input_shapes_;
+  std::vector<std::vector<TensorShape>> input_shapes_;
+
+  // Input shape values that we collect during profile_generation_mode. If the
+  // tensor is not compatible with a TRT shape tensor then an empty shape is
+  // stored.
+  std::vector<std::vector<nvinfer1::Dims>> input_shape_values_;
```

Comment:
Would this be better?
// The tensor values for the inputs with shape-tensor-compatible shapes. 
std::vector<std::vector<absl::optional<nvinfer1::dims>>>> maybe_shape_tensor_values;